blark/iec.lark

```diff
@@ -436,7 +439,7 @@ structured_var_init_decl: var1_list ":" initialized_structure
 
 // Function blocks
 fb_decl: fb_decl_name_list ":" function_block_type_name [ ":=" structure_initialization ] -> fb_name_decl
-       | fb_decl_name_list ":" function_call                                              -> fb_invocation_decl
+       | fb_decl_name_list ":" function_call [ ":=" structure_initialization ]            -> fb_invocation_decl
```

Comment:
This change means that `fb_invocation_decl` needs a third optional parameter for the `structure_initialization`, so https://github.com/engineerjoe440/blark/blob/e2dc9e80820466f952d7d1c6f2d67a6c6b0131fb/blark/transform.py#L1789 needs to be changed a bit (addition of the parameter + `__str__` changes).

If you take a look at this:
`pytest -v -k fb_initializer -s`
It'll show that the sample program gets reformatted as the following, entirely eating up the structure init:
`fbSample : FB_Sample(nInitParam := 1);`

Some of the parser tests are a bit "dumb" - so to speak - they might only check to see that the `.st` files parse and get transformed into dataclasses but not verify it produces the same code. The test suite could use improvement in that regard. 

Currently, `test_transformer` is the best place for that. Adding in a simple round-trip test of `fb_invocation_decl` with and without the structure initializer would be ideal, if you wouldn't mind!

Reply:
Not dumb at all - and that's the point of reviews! 👍

Reply:
I've made some relevant updates, but this has uncovered something else, that I'd really like your opinion (and scrutiny) on:

* https://github.com/engineerjoe440/blark/blob/bugfix/array-initializer/blark/iec.lark#L338-L339
* [blark/tests/test_transformer.py](https://github.com/klauer/blark/pull/56/files#diff-759678bb488ac11a49284f4850a8c3df75dbad9daeabadab516bb4b8c28e9516L357-L360)

Reply:
I _think_ this is OK but let me re-run blark on our codebases and make sure - will get back to you in a bit!

Reply:
Please and thank you! That'll make me feel a lot better.